queries.sql

```diff
@@ -61,5 +61,24 @@ SELECT a.name FROM animals a JOIN species s ON a.species_id = s.id JOIN owners o
 -- all animals owned by Dean Winchester that haven't tried to escape.
 SELECT a.name FROM animals a JOIN owners o ON a.owner_id = o.id WHERE o.full_name = 'Dean Winchester' AND a.escape_attempts = 0;
 
+-- Vet clinic database: add "join table" for visits
 -- Who owns the most animals?
 SELECT o.full_name, COUNT(*) AS animal_count FROM owners o JOIN animals a ON o.id = a.owner_id GROUP BY o.full_name ORDER BY animal_count DESC LIMIT 1;
+---Who was the last animal seen by William Tatcher?
+SELECT a.name FROM animals a JOIN visits v ON a.id = v.animal_id JOIN vets vet ON v.vet_id = vet.id WHERE vet.name = 'William Tatcher' ORDER BY v.date_of_visit DESC LIMIT 1;
+--How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT v.animal_id) AS animal_count FROM visits v JOIN vets vet ON v.vet_id = vet.id WHERE vet.name = 'Stephanie Mendez';
+--List all vets and their specialties, including vets with no specialties.
+SELECT v.name, s.name AS specialty FROM vets v LEFT JOIN specializations sp ON v.id = sp.vets_id LEFT JOIN species s ON sp.species_id = s.id;
+--List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name, v.date_of_visit FROM animals a JOIN visits v ON a.id = v.animal_id JOIN vets vet ON v.vet_id = vet.id WHERE vet.name = 'Stephanie Mendez' AND v.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+--What animal has the most visits to vets?
+SELECT a.name, COUNT(v.animal_id) AS visit_count FROM animals a JOIN visits v ON a.id = v.animal_id GROUP BY a.id ORDER BY visit_count DESC LIMIT 1;
+--Who was Maisy Smith's first visit?
+SELECT a.name, vs.date_of_visit, v.name FROM visits vs JOIN animals a ON a.id = vs.animal_id JOIN vets v ON v.id = vs.vet_id WHERE v.name = 'Maisy Smith' ORDER BY vs.date_of_visit LIMIT 1;
+--Details for most recent visit: animal information, vet information, and date of visit.
+SELECT a.name, vs.date_of_visit, v.name FROM visits vs JOIN animals a ON a.id = vs.animal_id JOIN vets v ON v.id = vs.vet_id ORDER BY vs.date_of_visit DESC LIMIT 1;
+--How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) AS not_specialty FROM visits v JOIN vets vet ON v.vet_id = vet.id JOIN animals a ON v.animal_id = a.id LEFT JOIN specializations sp ON vet.id = sp.vets_id AND a.species_id = sp.species_id WHERE sp.vets_id IS NULL;
+--What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT s.name AS suggested_specialty, COUNT(*) AS animal_count FROM animals a JOIN visits v ON a.id = v.animal_id JOIN vets vt ON v.vet_id = vt.id JOIN vets ms ON ms.name = 'Maisy Smith' JOIN specializations sp ON vt.id = sp.vets_id AND a.species_id = sp.species_id JOIN species s ON sp.species_id = s.id WHERE ms.id = vt.id GROUP BY s.name ORDER BY animal_count DESC LIMIT 1;
```

Comment:
- [ ] The result in the picture doesn't show the correct answer. What specialty should Maisy Smith consider getting? Look for the species she gets the most. I suggest grouping the animal visits by species, there are nine visits to Maisy Smith, three Pokemon, and six digimon.